git_theta/api.py

```diff
@@ -0,0 +1,74 @@
+"""User facing functions to interact with git-theta without incurring extra io costs."""
+
+import io
+import sys
+from typing import Optional, Union
+
+import git
+
+import git_theta
+from git_theta import checkpoints, filters, git_utils, metadata, utils
+
+
+def save(
+    state_dict,
+    path: str,
+    commit_msg: str,
+    checkpoint_type: str = "pytorch",
+    materialize: bool = False,
+) -> git.Commit:
+    """Save a model using git-theta without writing it to disk."""
+    repo = git_utils.get_git_repo()
+    # Convert the deep learning framework native state dict into our checkpoint format.
+    checkpoint_handler = checkpoints.get_checkpoint_handler(checkpoint_type)
+    ckpt = checkpoint_handler.from_framework(state_dict)
+    # Convert the checkpoint into the cleaned metadata file.
+    metadata = filters.clean(ckpt, repo, path)
+    # Capture metadata writing into a string.
+    with io.StringIO() as f:
+        metadata.write(f)
+        metadata = f.getvalue()
+    # Convert the metadata file into a git blob without having it on disk.
+    blob = git_theta.git_utils.make_blob(repo, metadata, path)
+    # Add the metadata to staging.
+    repo.index.add([blob])
+    # Commit the metadata.
+    if sys.platform in ("win32", "cygwin"):
+        # When you use GitPython to commit, things like hooks, i.e. our post-commit
+        # hook, run as subprocesses. Currently it seems that running shell scripts
+        # with the subprocess does not work in windows.
+        # Commit using the GitPython wrapper around the `git commit` command. This
+        # way hooks will be handled the same way as a normal commit.
+        repo.git.commit(m=commit_msg)
+        # We now need to get the sha manually in order to reference this commit.
+        sha = repo.commit("HEAD")
+    else:
+        # Commit directly from python
+        sha = repo.index.commit(commit_msg)
+    if materialize:
+        ckpt.save(path)
+    return sha
+
+
+def load(
+    sha_or_tag: Union[str, git.Commit],
+    path: str,
+    checkpoint_type: str = "pytorch",
+    checkout: bool = False,
+):
+    """Load a model from git-theta without having it checked out."""
```

Comment:
Probably change the docstring given the new checkout arg

Reply:
Updated